doc/adr/0012-modules-structure.md

```diff
@@ -0,0 +1,105 @@
+# 12. Modules' structure
+
+Date: 2022-12-28
+
+## Status
+
+Accepted
+
+## Context
+
+The existing structure of tessellation modules was mixed, hence hard to maintain.
+
+- `shared` contains both base classes and extensions, and tessellation schema,
```

Comment:
"contains base classes, extensions and tessellation schema" or 
"contains both base classes with extensions and tessellation schema"
not sure which one you meant

Reply:
fixed :+1: